aptos-move/framework/supra-framework/sources/multisig_voting.move

```diff
@@ -0,0 +1,1219 @@
+        execution_hash: vector<u8>,
+        expiration_secs: u64,
+        metadata: SimpleMap<String, vector<u8>>,
+        min_vote_threshold: u128,
+    }
+
+    struct RegisterForumEvent has drop, store {
+        hosting_account: address,
+        proposal_type_info: TypeInfo,
+    }
+
+    struct VoteEvent has drop, store {
+        proposal_id: u64
+    }
+
+    public fun register<ProposalType: store>(account: &signer) {
+        let addr = signer::address_of(account);
+        assert!(!exists<VotingForum<ProposalType>>(addr), error::already_exists(EVOTING_FORUM_ALREADY_REGISTERED));
+
+        let voting_forum = VotingForum<ProposalType> {
+            next_proposal_id: 0,
+            proposals: table::new<u64, Proposal<ProposalType>>(),
+            events: VotingEvents {
+                create_proposal_events: account::new_event_handle<CreateProposalEvent>(account),
+                register_forum_events: account::new_event_handle<RegisterForumEvent>(account),
+                resolve_proposal_events: account::new_event_handle<ResolveProposal>(account),
+                vote_events: account::new_event_handle<VoteEvent>(account),
+            }
+        };
+
+        if (std::features::module_event_migration_enabled()) {
+            event::emit(
+                RegisterForum {
+                    hosting_account: addr,
+                    proposal_type_info: type_info::type_of<ProposalType>(),
+                },
+            );
+        };
+        event::emit_event<RegisterForumEvent>(
+            &mut voting_forum.events.register_forum_events,
+            RegisterForumEvent {
+                hosting_account: addr,
+                proposal_type_info: type_info::type_of<ProposalType>(),
+            },
+        );
+
+        move_to(account, voting_forum);
+    }
+
+    /// Create a single-step proposal with the given parameters
+    ///
+    /// @param voting_forum_address The forum's address where the proposal will be stored.
+    /// @param execution_content The execution content that will be given back at resolution time. This can contain
+    /// data such as a capability resource used to scope the execution.
+    /// @param execution_hash The hash for the execution script module. Only the same exact script module can resolve
+    /// this proposal.
+    /// @param min_vote_threshold The minimum number of votes needed to consider this proposal successful.
+    /// @param expiration_secs The time in seconds at which the proposal expires and can potentially be resolved.
+    /// @param early_resolution_vote_threshold The vote threshold for early resolution of this proposal.
+    /// @param metadata A simple_map that stores information about this proposal.
+    /// @return The proposal id.
+    public fun create_proposal<ProposalType: store>(
+        proposer: address,
+        voting_forum_address: address,
+        execution_content: ProposalType,
+        execution_hash: vector<u8>,
+        min_vote_threshold: u128,
```

Comment:
`min_vote_threshold` must be greater than or equal to 2, this check must be present.

Reply:
okay, adding this checks in `create_proposal_v2` function, which is a general function